consensus/consensus.go

```diff
@@ -674,3 +770,81 @@ func (consensus *Consensus) GetNodeIDs() []libp2p_peer.ID {
 func (consensus *Consensus) GetConsensusID() uint32 {
 	return consensus.consensusID
 }
+
+// GenesisStakeInfoFinder is a stake info finder implementation using only
+// genesis accounts.
+// When used for block reward, it rewards only foundational nodes.
+type GenesisStakeInfoFinder struct {
+	byNodeKey map[types.BlsPublicKey][]*structs.StakeInfo
+	byAccount map[common.Address][]*structs.StakeInfo
```

Comment:
why one address has a list of stakeinfo?

Reply:
Good question.

Reply:
This is for the case where one node represents multiple ECDSA accounts, e.g. proxy staking.  The design came earlier from the staking discussion with you and Nicolas.  Right now we don’t have proxy staking so one BLS key always represents one account and vice versa, but it could change in the future.

Reply:
This is for the case where one node represents multiple ECDSA accounts, e.g. proxy staking.  The design came earlier from the staking discussion with you and Nicolas.  Right now we don’t have proxy staking so one BLS key always represents one account and vice versa, but it could change in the future.